Sources/Core/AWSClientRuntime/Middlewares/Route53TrimHostedZoneMiddleware.swift

```diff
@@ -0,0 +1,40 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import ClientRuntime
+
+public struct Route53TrimHostedZoneMiddleware<Input, Output>: ClientRuntime.Middleware {
```

Comment:
This is the same as the previously code-generated middleware, except:
- The input & output are generics that get substituted for each operation at compile time.
- The hosted zone ID on the input model is accessed via a key path that is passed in at initialization.